File: leet_724_find_pivot_index___naive.py
class Solution:
    def pivotIndex(self, n: List[int]) -> int:
        cur, s = 0, sum(n)
        for i in range(len(n)):
            if i > 0:
                cur += n[i - 1]
            if cur == (s - n[i]) / 2:
                return i
        return -1

File: leet_724_find_pivot_index___naive.java
class Solution {
    public int pivotIndex(int[] nums)
    {
        float   cur = 0;
        int     i = -1;
        int     s = 0;
        for (int n: nums)
            s += n;
        while (++i < nums.length)
        {
            if (i > 0)
                cur += nums[i - 1];
            if (cur == (s - n[i]) / 2)
                return i;
        }
        return -1;
    }
}
